conversion/conversion.go

```diff
@@ -543,33 +519,38 @@ func UpdateDatabase(project, instance, dbName string, conv *internal.Conv, out *
 		Statements: conv.SpSchema.GetDDL(ddl.Config{Comments: false, ProtectIds: true, Tables: true, ForeignKeys: false}),
 	})
 	if err != nil {
-		return "", fmt.Errorf("can't submit update schema request: %s", err)
+		return fmt.Errorf("can't build UpdateDatabaseDdlRequest: %w", AnalyzeError(err, project, instance))
 	}
 	if err := op.Wait(ctx); err != nil {
-		return "", fmt.Errorf("can't update schema statement: %s", err)
+		return fmt.Errorf("UpdateDatabaseDdl call failed: %w", AnalyzeError(err, project, instance))
 	}
 	fmt.Fprintf(out, "Updated schema successfully.\n")
-	return dbURI, nil
+	return nil
 }
 
-// UpdateDDLForeignKeys updates the Spanner database with foreign key
-// constraints using ALTER TABLE statements.
-func UpdateDDLForeignKeys(project, instance, dbName string, conv *internal.Conv, out *os.File) error {
-	ctx := context.Background()
-	adminClient, err := NewDatabaseAdminClient(ctx)
-	if err != nil {
-		return fmt.Errorf("can't create admin client: %w", analyzeError(err, project, instance))
+func splitDbURI(dbURI string) (project, instance, dbName string, err error) {
+	splitURI := strings.Split(dbURI, "/")
+	if len(splitURI) != 6 {
+		return "", "", "", fmt.Errorf("cannot split bad db URI: %s", dbURI)
+	}
```

Comment:
I think a better approach might be to restructure this function as following:

parseDbURI(dbURI string) (projectId, instanceId, dbName string) {
    split := strings.split("/databases/")
    instanceURI, dbName := split[0], split[1]
    projectId, instanceId := parseInstanceURI(instanceURI)
}

parseInstanceURI() (...) {
   ...
}